### PR TITLE
Add tab scrolling support for GTK3

### PR DESCRIPTION
**Pull request by tamplan**
I forgot mate-window-properties to handle his notebook and add mouse tab scrolling.

I forgot too a secondary notebook in mate-appearance-properties (customize theme window).

### CODE REVIEW

File: capplets/appearance/appearance-main.c [hunk 1]
```diff
@@ -204,9 +206,15 @@ main (int argc, char **argv)
   if (wallpaper_files && !install_filename && !start_page)
     start_page = g_strdup ("background");
 
-  GtkNotebook* nb = GTK_NOTEBOOK(appearance_capplet_get_widget (data, "main_notebook"));
+  nb = GTK_NOTEBOOK (appearance_capplet_get_widget (data, "main_notebook"));
   gtk_widget_add_events (GTK_WIDGET (nb), GDK_SCROLL_MASK);
-  g_signal_connect (GTK_WIDGET (nb), "scroll-event",
+  g_signal_connect (nb, "scroll-event",
+                    G_CALLBACK (capplet_dialog_page_scroll_event_cb),
+                    GTK_WINDOW (w));
```
Comment (rbuj) [inline capplets/appearance/appearance-main.c:213]:
```C
g_signal_connect(instance, detailed_signal, c_handler, data)
```
https://developer.gnome.org/gobject/stable/gobject-Signals.html#g-signal-connect
- instance = nb, it shoul be `GtkWidget*` bu it's `GtkNotebook*`, see 1st argument of `c_handler`
- c_handler = capplet_dialog_page_scroll_event_cb

```C
capplet_dialog_page_scroll_event_cb (GtkWidget *widget, GdkEventScroll *event, GtkWindow *window)
```

File: capplets/appearance/appearance-main.c [hunk 2]
```diff
@@ -204,9 +206,15 @@ main (int argc, char **argv)
...
+
+  nb_custom_theme = GTK_NOTEBOOK(appearance_capplet_get_widget (data, "notebook2"));
+  gtk_widget_add_events (GTK_WIDGET (nb_custom_theme), GDK_SCROLL_MASK);
+  g_signal_connect (nb_custom_theme, "scroll-event",
                     G_CALLBACK (capplet_dialog_page_scroll_event_cb),
                     GTK_WINDOW (w));
```
Comment (rbuj) [inline capplets/appearance/appearance-main.c:219]: It should be called after:
```
g_option_context_free (option_context);
```
Don't cast GtkWidget* to GtkNotebook*:
```C
capplets/appearance/appearance.h:#define appearance_capplet_get_widget(x, y) (GtkWidget*) gtk_builder_get_object(x->ui, y)
```
```C
nb_custom_theme = appearance_capplet_get_widget (data, "notebook2");
gtk_widget_add_events (nb_custom_theme, GDK_SCROLL_MASK);
```

Comment (tamplan) [reply capplets/appearance/appearance-main.c:219]: Same code in "appearance-main.c" than "mate-window-properties.c"

Comment (rbuj) [reply capplets/appearance/appearance-main.c:219]: nb is used on argument parsing, nb_custom_theme is not.

gint gtk_notebook_page_num (GtkNotebook *notebook, GtkWidget *child);
void gtk_notebook_set_current_page (GtkNotebook *notebook, gint page_num);

https://github.com/mate-desktop/mate-control-center/blob/3c7d68bdee43165d963a340f58ec4ea8c4ac7449/capplets/appearance/appearance-main.c#L221-L236

Comment (tamplan) [reply capplets/appearance/appearance-main.c:219]: nb you speak is local in function 'main_window_response' and stay untouch.

Comment (tamplan) [reply capplets/appearance/appearance-main.c]: updated just now

Comment (rbuj) [reply capplets/appearance/appearance-main.c]: compiler warnings:
```
$ ./autogen.sh --enable-compile-warnings=maximum --prefix=/usr && make
<cut>
appearance-main.c: In function ‘main’:
appearance-main.c:231:39: warning: passing argument 1 of ‘gtk_notebook_page_num’ from incompatible pointer type [-Wincompatible-pointer-types]
  231 |       pindex = gtk_notebook_page_num (nb, w);
      |                                       ^~
      |                                       |
      |                                       GtkWidget * {aka struct _GtkWidget *}
In file included from /usr/include/gtk-3.0/gtk/gtk.h:155,
                 from appearance.h:24,
                 from appearance-main.c:23:
/usr/include/gtk-3.0/gtk/gtknotebook.h:186:56: note: expected ‘GtkNotebook *’ {aka ‘struct _GtkNotebook *’} but argument is of type ‘GtkWidget *’ {aka ‘struct _GtkWidget *’}
  186 | gint       gtk_notebook_page_num         (GtkNotebook *notebook,
      |                                           ~~~~~~~~~~~~~^~~~~~~~
appearance-main.c:233:40: warning: passing argument 1 of ‘gtk_notebook_set_current_page’ from incompatible pointer type [-Wincompatible-pointer-types]
  233 |         gtk_notebook_set_current_page (nb, pindex);
      |                                        ^~
      |                                        |
      |                                        GtkWidget * {aka struct _GtkWidget *}
<cut>
```

Comment (tamplan) [reply capplets/appearance/appearance-main.c]: ok, give a lot of time to test here...

Comment (tamplan) [reply capplets/appearance/appearance-main.c]: Should be correct now.

File: capplets/appearance/appearance-main.c [hunk 3]
```diff
@@ -204,9 +206,15 @@ main (int argc, char **argv)
...
+  gtk_widget_add_events (nb, GDK_SCROLL_MASK);
+  g_signal_connect (nb, "scroll-event",
+                    G_CALLBACK (capplet_dialog_page_scroll_event_cb),
+                    GTK_WINDOW (w));
+
+  nb_custom_theme = appearance_capplet_get_widget (data, "notebook2");
+  gtk_widget_add_events (nb_custom_theme, GDK_SCROLL_MASK);
+  g_signal_connect (nb_custom_theme, "scroll-event",
                     G_CALLBACK (capplet_dialog_page_scroll_event_cb),
                     GTK_WINDOW (w));
```
Comment (rbuj) [inline capplets/appearance/appearance-main.c:219]: Move this code block after the line:
```
  g_option_context_free (option_context);
```

File: capplets/appearance/appearance-main.c [hunk 4]
```diff
@@ -220,9 +228,9 @@ main (int argc, char **argv)
     if (w != NULL) {
       gint pindex;
 
-      pindex = gtk_notebook_page_num (nb, w);
+      pindex = gtk_notebook_page_num (GTK_NOTEBOOK (nb), w);
       if (pindex != -1)
-        gtk_notebook_set_current_page (nb, pindex);
+        gtk_notebook_set_current_page (GTK_NOTEBOOK (nb), pindex);
```
Comment (rbuj) [inline capplets/appearance/appearance-main.c:233]: if you use `GTK_NOTEBOOK (nb)` more than once, you can assign it to a local variable, for example within the scope of the if block.

Comment (tamplan) [reply capplets/appearance/appearance-main.c]: Like this?

**Comment by rbuj** [comment]
```C
  /* default to background page if files were given on the command line */
  if (wallpaper_files && !install_filename && !start_page)
    start_page = g_strdup ("background");

  nb = appearance_capplet_get_widget (data, "main_notebook");
  gtk_widget_add_events (nb, GDK_SCROLL_MASK);
  g_signal_connect (nb, "scroll-event",
                    G_CALLBACK (capplet_dialog_page_scroll_event_cb),
                    GTK_WINDOW (w));

  if (start_page != NULL) {
    gchar *page_name;

    page_name = g_strconcat (start_page, "_vbox", NULL);
    g_free (start_page);

    w = appearance_capplet_get_widget (data, page_name);
    if (w != NULL) {
      gint pindex;
      GtkNotebook *notebook;

      notebook = GTK_NOTEBOOK (nb);
      pindex = gtk_notebook_page_num (notebook, w);
      if (pindex != -1)
        gtk_notebook_set_current_page (notebook, pindex);
    }
    g_free (page_name);
  }

  if (install_filename != NULL) {
    GFile *inst = g_file_new_for_commandline_arg (install_filename);
    g_free (install_filename);
    mate_theme_install (inst, GTK_WINDOW (w));
    g_object_unref (inst);
  }

  g_option_context_free (option_context);

  nb_custom_theme = appearance_capplet_get_widget (data, "notebook2");
  gtk_widget_add_events (nb_custom_theme, GDK_SCROLL_MASK);
  g_signal_connect (nb_custom_theme, "scroll-event",
                    G_CALLBACK (capplet_dialog_page_scroll_event_cb),
                    GTK_WINDOW (w));
```

**Comment by tamplan** [comment]
Done, sorry for the time wasted...

**Pull request merged**